src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -3595,22 +3585,28 @@ void MarkInterfacesNeededByBodyStack (MethodBody body)
 				MarkInterfaceImplementation (implementation, new MessageOrigin (type));
 		}
 
+		bool InstructionRequiresReflectionMethodBodyScannerForFieldAccess (Instruction instruction)
+			=> instruction.OpCode.Code switch {
+				// Field stores (Storing value to annotated field must be checked)
+				Code.Stfld or
+				Code.Stsfld or
+				// Field address loads (as those can be used to store values to annotated field and thus must be checked)
+				Code.Ldflda or
+				Code.Ldsflda
+					=> ReflectionMethodBodyScanner.RequiresReflectionMethodBodyScannerForAccess (Context, (FieldReference) instruction.Operand),
+				// For ref fields, ldfld loads an address which can be used to store values to annotated fields
+				Code.Ldfld when ((FieldReference) instruction.Operand).FieldType.IsByRefOrPointer ()
```

Comment:
I don't think there's an easy way to tell where the value is being stored or what value we're storing to it since stind operates on addresses on the stack. And this will filter to only be ref fields or pointers that also have annotations, so it shouldn't cause too many more method bodies to be analyzed for dataflow.